### PR TITLE
Make sure that loggers can be enabled, disabled, captured and tee'd from MinimalLib

### PR DESCRIPTION
**Pull request by ptosco**
I noticed that capturing/tee'ing loggers from MinimalLib was not as reliable as it should be, so I made it more robust and added a good amount of tests on both CFFI and MinimalLib sides.

### CODE REVIEW

**Comment by bp-kelley** [comment]
@ptosco I'm going to take a bit with this one if you don't mind.  If you don't hear from me by Monday Nov 11 ping me again

File: Code/MinimalLib/cffi_test.c [hunk 1]
```diff
@@ -2170,6 +2176,160 @@ void test_partial_sanitization() {
...
+}
+
+CapturedStreams *capture_streams(unsigned int buf_size) {
+  CapturedStreams *res;
+  res = (CapturedStreams *)malloc(sizeof(CapturedStreams));
+  if (!res) {
+    return NULL;
+  }
+  memset(res, 0, sizeof(CapturedStreams));
+  res->stdout_pipes[0] = -1;
```
Comment (bp-kelley) [inline Code/MinimalLib/cffi_test.c:2236]: suggest using
```
for(int i=0;i<PIPE_FD_SIZE; ++i) {
   res->stdout_pipes[i] = -1;
   res->stderr_pipes[i] = -1;
}
```

File: Code/MinimalLib/common.h [hunk 2]
```diff
@@ -1090,26 +1094,159 @@ std::string parse_inchi_options(const char *details_json) {
 }
 #endif
 
-struct LogHandle {
+namespace {
+#ifdef RDK_BUILD_THREADSAFE_SSS
+std::mutex &logger_mutex_get() {
+  // create on demand
```
Comment (bp-kelley) [inline Code/MinimalLib/common.h:1100]: If this is C++11 or above you only need the logger_mutex_get, the static creation of _mutex will be thread-safe initialized and not require std::call_once

https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/n4594.pdf

From the section 6.7:

such a variable is initialized the first time control passes through its declaration; such a variable is considered initialized upon the completion of its initialization. [...] If control enters the declaration concurrently while the variable is being initialized, the concurrent execution shall wait for completion of the initialization.

Then there's a footnote:

The implementation must not introduce any deadlock around execution of the initializer.

File: Code/MinimalLib/cffi_test.c (hunk 1 above)
Comment (ptosco) [reply Code/MinimalLib/cffi_test.c:2236]: Done.

File: Code/MinimalLib/common.h (hunk 2 above)
Comment (ptosco) [reply Code/MinimalLib/common.h:1100]: We have the same code elsewhere in RDKit (e.g., in `Code/GraphMol/DistGeomHelpers/Embedder.cpp` and `Code/GraphMol/MolPickler.cpp`); I would rather leave it as is here and then submit a separate PR to remove `std::call_once` if we believe that is unnecessary.

Comment (bp-kelley) [reply Code/MinimalLib/common.h:1139]: Back then it was necessary, now it is not.  For new code we should use the current spec IMO

Comment (bp-kelley) [reply Code/MinimalLib/common.h:1139]: As it makes code a lot simpler

Comment (bp-kelley) [reply Code/MinimalLib/common.h:1139]: But the code is correct in both cases so I won’t fight this one

Comment (ptosco) [reply Code/MinimalLib/common.h:1154]: @bp-kelley Hi Brian, I have applied your suggestion and removed `std::call_once` from this PR.
Are you OK to merge this PR now or do you need more time?

Comment (ptosco) [reply Code/MinimalLib/common.h:1154]: @bp-kelley Gentle poke: I have applied your suggestion and removed std::call_once from this PR.
Are you OK to merge now or do you need more time?

Comment (bp-kelley) [reply Code/MinimalLib/common.h:1154]: I admit to wishing there was an easier way to do this, but it finally makes sense to me.

LGTM

**Review by bp-kelley — APPROVED** [review]

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**